crates/router/src/connector/cybersource/transformers.rs

```diff
@@ -1103,30 +1292,71 @@ impl<F> TryFrom<&CybersourceRouterData<&types::RefundsRouterData<F>>> for Cybers
     }
 }
 
-impl TryFrom<types::RefundsResponseRouterData<api::Execute, CybersourcePaymentsResponse>>
+impl From<CybersourceRefundStatus> for enums::RefundStatus {
+    fn from(item: CybersourceRefundStatus) -> Self {
+        match item {
+            CybersourceRefundStatus::Succeeded | CybersourceRefundStatus::Transmitted => {
+                Self::Success
+            }
+            CybersourceRefundStatus::Failed | CybersourceRefundStatus::Voided => Self::Failure,
+            CybersourceRefundStatus::Pending => Self::Pending,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum CybersourceRefundStatus {
+    Succeeded,
+    Transmitted,
+    Failed,
+    Pending,
+    Voided,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CybersourceRefundResponse {
+    id: String,
+    status: CybersourceRefundStatus,
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::Execute, CybersourceRefundResponse>>
     for types::RefundsRouterData<api::Execute>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::RefundsResponseRouterData<api::Execute, CybersourcePaymentsResponse>,
+        item: types::RefundsResponseRouterData<api::Execute, CybersourceRefundResponse>,
     ) -> Result<Self, Self::Error> {
-        let refund_status = enums::RefundStatus::from(item.response.status);
         Ok(Self {
             response: Ok(types::RefundsResponseData {
                 connector_refund_id: item.response.id,
-                refund_status,
+                refund_status: enums::RefundStatus::from(item.response.status),
             }),
             ..item.data
         })
     }
 }
 
-impl TryFrom<types::RefundsResponseRouterData<api::RSync, CybersourceTransactionResponse>>
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RsyncApplicationInformation {
+    status: CybersourceRefundStatus,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CybersourceRsyncResponse {
```

Comment:
Since we have handled ErrorResponse for authorize, cancel, capture, psync. do we need to consider the same for refund and refund sync?

Reply:
This is not required for refunds and rsync because they always send status in the response